cmd/api/user_test.go

```diff
@@ -0,0 +1,927 @@
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusOK, w.Code, "HTTP Status Code")
+		var users []versionary.TextValue
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&users), "Decode JSON Users") {
+			expect.Equal(1, len(users), "Number of Users")
+		}
+	}
+	// Get user IDs and names: happy path (sorted parameters)
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_names?sorted=true", nil)
+	req.Header.Set("Authorization", "Bearer "+adminToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusOK, w.Code, "HTTP Status Code")
+		var users []versionary.TextValue
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&users), "Decode JSON Users") {
+			names := versionary.Map(users, func(e versionary.TextValue) string { return e.Value })
+			knownNames := versionary.Map(knownUsers, func(u user.User) string { return u.String() })
+			expect.Equal(2, len(users), "Number of Users")
+			expect.Equal(names[1], knownNames[0], "User Name")
+		}
+	}
+}
+
+func TestReadUserEmails(t *testing.T) {
+	expect := assert.New(t)
+	// Get users emails: missing authentication token
+	w := httptest.NewRecorder()
+	req, err := http.NewRequest("GET", "/v1/user_emails", nil)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusUnauthorized, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusUnauthorized, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthenticated", "Event Message")
+		}
+	}
+	// Get users emails: unauthorized token (not an admin)
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_emails", nil)
+	req.Header.Set("Authorization", "Bearer "+regularToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusForbidden, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusForbidden, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthorized", "Event Message")
+		}
+	}
+	// Get users emails: happy path
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_emails", nil)
+	req.Header.Set("Authorization", "Bearer "+adminToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusOK, w.Code, "HTTP Status Code")
+		var emails []string
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&emails), "Decode JSON Users") {
+			expect.Equal(3, len(emails))
```

Comment:
We can't guarantee that the CRUD test user exists when this test runs. If you run this test by itself, it will fail. So, we should use "greater or equal to".